src/test/java/io/appium/java_client/android/FingerPrintTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android;
+
+import io.appium.java_client.remote.MobileCapabilityType;
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.util.concurrent.TimeUnit;
+
+public class FingerPrintTest {
+    private static AppiumDriverLocalService service;
+
+    /**
+     * initialization.
+     */
+    @BeforeClass public static void beforeClass() throws Exception {
+        service = AppiumDriverLocalService.buildDefaultService();
+        service.start();
+
+        if (service == null || !service.isRunning()) {
+            throw new RuntimeException("An appium server node is not started!");
+        }
+    }
+
+    /**
+     * finishing.
+     */
+    @AfterClass public static void afterClass() {
+        if (service != null) {
+            service.stop();
+        }
+    }
+
+    /**
+     * enable system security which is required for finger print activation.
+     */
+    @Before public void before() throws Exception {
+        final AndroidDriver driver = getAndroidDriver("ChooseLockGeneric");
+        TimeUnit.SECONDS.sleep(2);
+        // clicking the pin lock mode
+        driver.findElement(By.xpath("//android.widget.ListView[1]/android.widget.LinearLayout[4]"))
+            .click();
+        TimeUnit.SECONDS.sleep(2);
+        driver.findElement(By.id("com.android.settings:id/password_entry")).sendKeys("1234\n");
+        TimeUnit.SECONDS.sleep(2);
+        driver.findElement(By.id("com.android.settings:id/password_entry")).sendKeys("1234\n");
+        TimeUnit.SECONDS.sleep(2);
+        driver.findElement(By.id("com.android.settings:id/next_button")).click();
+        TimeUnit.SECONDS.sleep(2);
+        driver.quit();
+    }
+
+    /**
+     * add a new finger print to security.
+     */
+    @Test public void pressKeyCodeTest() throws Exception {
```

Comment:
@SrinivasanTarget the code is suppose to throw exception is the test fails. I add try catch to whole test body and assert fail on failing :-)

Reply:
use the `@Test(expected = MyException.class)` syntax for such case

Reply:
@mykola-mokhnach throwing exception is not the expected behavior. if it throws any that means it failed. try catch is there to prevent throwing exception on test fail (as asked by @SrinivasanTarget )

Reply:
ok, perhaps it makes sense to add some nice description to the `fail` method in such case

Reply:
Done